src/actions/commands/insert.ts

```diff
@@ -284,6 +284,32 @@ export class CommandBackspaceInInsertMode extends BaseCommand {
   }
 }
 
+@RegisterAction
+export class CommandDeleteInInsertMode extends BaseCommand {
+  modes = [Mode.Insert];
+  keys = ['<Del>'];
+  mightChangeDocument = true;
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    const selection = TextEditor.getSelection();
+
+    if (!selection.isEmpty) {
+      // If a selection is active, delete it
+      vimState.recordedState.transformations.push({
+        type: 'deleteRange',
+        range: new Range(selection.start as Position, selection.end as Position),
+      });
+    } else if (!position.isAtDocumentEnd()) {
+      // Otherwise, just delete a character (unless we're at the end of the document)
+      vimState.recordedState.transformations.push({
+        type: 'deleteText',
+        position: position.getRightThroughLineBreaks(true),
```

Comment:
I was surprised that this is vim's default behavior when I double-checked. I then wondered if it was configurable, which led me to file https://github.com/VSCodeVim/Vim/issues/4573